### PR TITLE
Badger keymaps

### PR DESCRIPTION
**Pull request by BlueTufa**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/1upkeyboards/1up60hte/keymaps/badger/keymap.c [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+     _______,    MM_LH,    MM_MAX,   MM_RH,    IJ_FIND,  IJ_IMPS,  KC_LEFT,  KC_DOWN,  KC_UP,    KC_RGHT,  CM_RIGHT, CM_DOWN,   _______,  \
+     _______,    MM_UH,    MM_BH,    MAC_CPY,  MAC_PST,  IJ_IMPH,  _______,  IJ_REN,   IJ_IMPL,  IJ_DECL,  IJ_USAG,  _______,   _______,  \
+     _______,    _______,  _______,  _______,  _______,  _______,  _______),
+
+  [_QWERTY_LINUX] = LAYOUT_tsangan(\
+     KC_GRV,     KC_1,     KC_2,     KC_3,     KC_4,     KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,     KC_MINS,   KC_EQL,   KC_BSPC, KC_DEL, \
+     KC_TAB,     KC_Q,     KC_W,     KC_E,     KC_R,     KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,     KC_LBRC,   KC_RBRC,  KC_BSLS, \
+     MOVE,       KC_A,     KC_S,     KC_D,     KC_F,     KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,  KC_QUOT,   KC_ENT,   \
+     KC_LSFT,    KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,  KC_RSFT,   CFG_LNX, \
+     KC_LCTL,    KC_LGUI,  KC_LALT,  KC_SPC,   ADJUST,   KC_RALT,  KC_RGUI),
```
Comment (drashna) [inline keyboards/1upkeyboards/1up60hte/keymaps/badger/keymap.c:23]: I'm not sure about the rest of your keymap, but if the only changes are alt and gui, then you could actually just use bootmagic. 

https://docs.qmk.fm/#/feature_bootmagic?id=keycodes

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Also, could you add a license header to each of the C and H files, here?

**Review by BlueTufa — COMMENTED** [review]
> Also, could you add a license header to each of the C and H files, here?

Done.

File: keyboards/1upkeyboards/1up60hte/keymaps/badger/keymap.c (hunk 1 above)
Comment (BlueTufa) [reply keyboards/1upkeyboards/1up60hte/keymaps/badger/keymap.c:37]: I do occasionally use the Bootmagic swap, but the big change is that I have 2 layers each with a set of os-specific shortcuts(move_mac, move_linux).  I am always looking for a cleaner solution than this, although I find it pretty manageable despite the fact that I keep trying new 60% boards with slightly different layouts.  If I come up with something interesting I'll open a PR.

Comment (drashna) [reply keyboards/1upkeyboards/1up60hte/keymaps/badger/keymap.c:37]: Okay, sounds good. 

Though, as an idea, you could look at the bootmagic/keymap config, and have the macros work depending on what mode it is in.   You don't have to d that here, but just some "food for thought".

**Review by noroadsleft — CHANGES_REQUESTED** [review]
GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/10239/files).

File: keyboards/1upkeyboards/1up60rgb/1up60rgb.h [hunk 2]
```diff
@@ -33,6 +33,21 @@
     { k40, k41, XXX, k43, XXX, XXX, k46, XXX, XXX, XXX, k4A, k4B, XXX, k4D, k4E } \
 }
 
+/* ANSI-Tsangan variant, with split right shift key */
+#define LAYOUT_60_ansi_tsangan( \
```
[the remark this answers is not in the document]
Comment (BlueTufa) [reply keyboards/1upkeyboards/1up60rgb/1up60rgb.h]: Makes sense now.  I saw that in `layouts/default/60_ansi_tsangan/info.json` then noticed there were keyboards using it with 2.75u space.

File: keyboards/1upkeyboards/1up60rgb/rules.mk [hunk 3]
```diff
@@ -25,4 +25,4 @@ BACKLIGHT_ENABLE = yes  # Enable keyboard backlight functionality
 AUDIO_ENABLE = no
 RGBLIGHT_ENABLE = yes
 
-LAYOUTS = 60_ansi 60_iso 60_ansi_split_bs_rshift 60_hhkb
+LAYOUTS = 60_ansi 60_iso 60_ansi_split_bs_rshift 60_hhkb 60_ansi_tsangan 60_ansi_split_rshift
```
[the remark this answers is not in the document]
Comment (BlueTufa) [reply keyboards/1upkeyboards/1up60rgb/rules.mk]: Thank you for the code review, this was very educational!

File: keyboards/1upkeyboards/1up60rgb/keymaps/badger/new_tsangan.c [hunk 4]
```diff
@@ -0,0 +1,59 @@
+/*
```
Comment (noroadsleft) [inline keyboards/1upkeyboards/1up60rgb/keymaps/badger/new_tsangan.c:1]: This file actually can't be compiled due to the file name (keymap files must be named either `keymap.c` or `keymap.json`).

You could move this to `keyboards/1upkeyboards/1up60rgb/keymaps/badger-tsangan/keymap.c`, and then add a `keyboards/1upkeyboards/1up60rgb/keymaps/badger-tsangan/rules.mk` that contains `USER_NAME := badger`, and then compile either keymap as desired. (I'm thinking you have two of this board - one Tsangan bottom row and one standard, yes?)

Comment (BlueTufa) [reply keyboards/1upkeyboards/1up60rgb/keymaps/badger/new_tsangan.c]: I addressed this by making the current keymap the default.  

I did order a second one of these boards, but it's been delayed for a while due to the current conditions.  Currently I've been testing my layouts by resoldering the bottom row :)

**Review by noroadsleft — APPROVED** [review]

File: keyboards/1upkeyboards/1up60rgb/keymaps/badger/new_tsangan.c (hunk 4 above)
Comment (noroadsleft) [reply keyboards/1upkeyboards/1up60rgb/keymaps/badger/new_tsangan.c]: > Lately I've been testing my layouts by resoldering the bottom row :)

That strikes me as tedious, but you do you. :)

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**